### PR TITLE
add page templates

### LINKED ISSUE #15: Stub out initial pages
## Description
Set up case study, resource and index page types

## Plan
- [x] Make an index/home page with a title
- [x] Make a case study page with title
- [x] Make a resource with a title

### PR DESCRIPTION
**Pull request by r-ferrier**
Fixes #15 

## Description

- Adds three initial pages - index, resource, case study
- I've added in separate titles for each and removed the index view from Main.elm and into Page/Index
- I've added very basic routing for each

If this is a weird pattern I've used or anyone has any thoughts on naming these differently please let me know!  I couldn't work out yet how we would reach these from the Main.elm but figured once we sort out routing properly we will sort that out then. 

@geeksforsocialchange/developers

### CODE REVIEW

File: src/I18n/Cy.elm [hunk 1]
```diff
@@ -10,7 +10,13 @@ cyStrings key =
             "[cCc] Init template CY"
 
         SiteTitle ->
-            "Team Wilder CY"
+            "[cCc] Team Wilder CY"
+
+        CaseStudyTitle ->
+            "[cCc] Case Study CY"
```
Comment (katjam) [inline src/I18n/Cy.elm:16]: We'll move this to markdown during Alpha milestone.

File: src/Main.elm [hunk 2]
```diff
@@ -22,7 +21,7 @@ main =
         { init = init
         , update = update
         , subscriptions = subscriptions
-        , view = viewDocument
+        , view = Page.Index.view
```
Comment (katjam) [inline src/Main.elm:24]: I think we want to keep this as viewDocument and pass the page as an parameter.

File: src/Page/CaseStudy.elm [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+import Browser
+import Html.Styled exposing (toUnstyled)
+import I18n.Keys exposing (Key(..))
+import I18n.Translate exposing (translate)
+import Shared exposing (Model, Msg)
+import Theme.PageTemplate as PageTemplate
+
+
+view : Model -> Browser.Document Msg
```
Comment (katjam) [inline src/Page/CaseStudy.elm:11]: The view function for each page should return `Html Msg`

File: src/Route.elm [hunk 4]
```diff
@@ -20,6 +22,12 @@ toString route =
         Index ->
             "/"
 
+        CaseStudy ->
```
Comment (katjam) [inline src/Route.elm:25]: This is great as a first step. We'll want it to take a parameter and be `case-study/title`.

File: src/Route.elm [hunk 5]
```diff
@@ -20,6 +22,12 @@ toString route =
...
+            "/case-study"
```
Comment (katjam) [inline src/Route.elm:26]: I think we probably want to call this route something other than case-study. I'm speaking to @undividual about it here #36

File: src/Theme/PageTemplate.elm [hunk 6]
```diff
@@ -1,43 +1,38 @@
...
-    case model.page of
-        Index ->
-            div []
-                [ p [] [ text (t PageTitle) ]
-                , button [ onClick LanguageChangeRequested ] [ text (t ChangeLanguage) ]
-                ]
+    div [ css [ defaultPageStyle ] ]
+        [ HeaderTemplate.view { content = t pageInfo.title }
+        , main_ [ css [ mainStyle ] ]
+            [ button [ onClick LanguageChangeRequested ] [ text (t ChangeLanguage) ]
```
Comment (katjam) [inline src/Theme/PageTemplate.elm:32]: We're going to move this, but not important now. I think the shape of things will make more sense when we get a few more features in.

**Review by katjam — COMMENTED** [review]
This is looking pretty good, but let's talk briefly tomorrow. I think we're over complicating again.

I'll try to find a simple example to model on since I'm not around much next few days.

The most important simplification is to get the Browser.Document Msg type retun back into only main since it's the end point. Every v`view` function should return Html Msg by convention and also to keep things compatible when passing around.

File: src/Page/CaseStudy.elm (hunk 3 above)
Comment (aaaaargZombies) [reply src/Page/CaseStudy.elm:11]: duplicate line here too I think.

File: src/Route.elm [hunk 7]
```diff
@@ -20,6 +22,12 @@ toString route =
...
+
+        Resource ->
+            "/resource"
+
 
 routeParser : Parser (Route -> a) a
 routeParser =
```
Comment (aaaaargZombies) [inline src/Route.elm:33]: can't seem to add a comment to all the lines but we probably want to read the routes here.

```elm
routeParser : Parser (Route -> a) a
routeParser =
    oneOf
        [ map Index top
        , map Resource (s "resource")
        , map CaseStudy (s "case-study")
        ]
```

https://package.elm-lang.org/packages/elm/url/latest/Url-Parser

That should then update the `model.page` when we load.

File: src/Main.elm (hunk 2 above)
Comment (aaaaargZombies) [reply src/Main.elm:24]: So now we have the `routeParser` (I hope these comments come in order!?) as Katja says we can match on the `page`

something like 

```elm
viewDocument : Model -> Browser.Document Msg
viewDocument model =
    case model.page of
        Index ->
            Page.Index.view model

        CaseStudy ->
            Page.CaseStudy.view model

        Resource ->
            Page.Resource.view model
```

Comment (r-ferrier) [reply src/Main.elm:24]: Nice!

File: src/Route.elm (hunk 5 above)
Comment (katjam) [reply src/Route.elm:26]: No, just put it in like this - we'll make the decision in that other ticket.

**Comment by r-ferrier** [comment]
Ok I've made those changes @katjam and @aaaaargZombies , I've just rolled back the json changes locally to test it's all working I think. I was slightly unsure about the route parsing so let me know if that looks wrong.

**Review by aaaaargZombies — APPROVED** [review]
working nicely!

I think the `toString` function is not quite what we want but as it's not actually used anywhere let's merge and deal with it when it matters.

**Pull request merged**